doc/api/addons.markdown

```diff
@@ -85,37 +105,96 @@ to build your module in a JSON-like format. This file gets compiled by
-handles, scopes, function templates, etc.
-
-In order to use these examples, you need to compile them using `node-gyp`.
-Create the following `binding.gyp` file:
+Because the exact path to the compiled Addon binary can vary depending on how
+it is compiled (i.e. sometimes it may be in `./build/Debug/`), Addons can use
+the [bindings][] package to load the compiled module.
+ 
+### Linking to Node.js' own dependencies
+
+Node.js uses a number of statically linked libraries such as v8, libuv and
+OpenSSL. All Addons are required to statically link to v8 and may link to any
+of the other dependencies as well. Typically, this is as simple as including
+the appropriate `#include <...>` statements (e.g. `#include <v8.h>`) and
+`node-gyp` will locate the appropriate headers automatically. However, there
+are a few caveats to be aware of:
+
+* When `node-gyp` runs, it will detect the specific release version of Node.js
+and download either the full source tarball or just the headers. If the full
+source is downloaded, Addons will have complete access to the full set of
+Node.js dependencies. However, if only the Node.js headers are downloaded, then only the symbols exported by Node.js will be available.
+* `node-gyp` can be run using the `--nodedir` flag pointing at a local Node.js
+source image. Using this option, the Addon will have access to the full set of
+dependencies.
+
+### Loading Addons using require()
+
+The filename extension of the compiled Addon binary is `.node` (as opposed
+to `.dll` or `.so`). The `require()` function is written to look for files
+with the `.node` file extension and initialize those as dynamically-linked
+libraries.
+
+When calling `require()`, the `.node` extension can usually be
+omitted and Node.js will still find and initialize the Addon. One caveat,
+however, is that Node.js will first attempt to locate and load modules or
+JavaScript files that happen to share the same base name. For instance, if
+there is a file `addon.js` in the same directory as the binary `addon.node`,
+then `require('addon')` will give precedence to the `addon.js` file and load it
+instead.
+
+## Native Abstractions for Node.js
+
+Each of the examples illustrated in this document make direct use of the
+Node.js and V8 APIs for implementing Addons. It is important to understand
+that the V8 API can, and has, changed dramatically from one V8 release to the
+next (and one major Node.js release to the next). With each change, Addons may
+need to be updated and recompiled in order to continue functioning. The Node.js
+release schedule is designed to minimize the frequency and impact of such
+changes but there is little that Node.js can do currently to ensure stability
+of the V8 APIs.
+
+The [Native Abstrations for Node.js][] (or `nan`) provide a set of tools that
+Addon developers can use to more easily adapt to changes in the V8 APIs. See
+the `nan` [examples][] for an illustration of how it can be used.
```

Comment:
Given that any native module that does not use nan will be problematic for the community, how about we word these differently?

How about?

The [Native Abstrations for Node.js][] (or `nan`) provide a set of tools that
Addon developers are __recommended to use__ to keep compatibility between past and future releases of V8 and Node.js. See
the `nan` [examples][] for an illustration of how it can be used. 

[Native Abstrations for Node.js]: https://github.com/nodejs/nan
[examples]: https://github.com/nodejs/nan/tree/master/examples/

However, I would probably go further: given that nan is within by the Node.js Foundation, how about we use it in our examples?